### ISSUE TITLE
Since there is no image that contains a bounding box, cannot display any image.

### ISSUE THREAD

**Issue by mdziec** [sassoftware/python-dlpy#86]
What could be the reason for the following message?

Since there is no image that contains a bounding box, cannot display any image.

from 
predict_tbl = ImageTable.load_files(conn=sess, path='/opt/sasinside/DemoData/test_img2/')
predict_tbl.resize(height=416, width=416, inplace=True)
model.predict(data=predict_tbl)
display_object_detections(conn=sess, 
                          coord_type='yolo', 
                          max_objects=5, 
                          table=model.valid_res_tbl, 
                          num_plot=10,
                          n_col=2)

Even if I use the image on which the model was trained, I do not receive any recognized object

**Comment by sasxianhu** [comment]
This usually means your model is not converged.

**Comment by bensloane** [comment]
Are you using the provided weights or training from scratch? Could be that your model has not been trained long enough. Do you have any fit statistics or metrics?

**Comment by mdziec** [comment]
How can I get to know it?

solver = MomentumSolver(learning_rate=0.001, clip_grad_max = 100, clip_grad_min = -100)
optimizer = Optimizer(algorithm=solver, mini_batch_size=10, log_level=2, max_epochs=20, reg_l2=0.005)
data_specs = [DataSpec(type_='IMAGE', layer='Input1', data=inputVars),
              DataSpec(type_='OBJECTDETECTION', layer='Detection1', data=targets)]
#gpu = Gpu(devices=[0])
model.set_weights('TinyYoloV2Demo')
model.fit(data=dettbl, 
               optimizer=optimizer, 
               data_specs=data_specs, 
               n_threads=20, 
               record_seed=13309,
               force_equal_padding=True,
          save_best_weights=True
               )

I tried to change the number of epochs from 10 to 200 without effect.

  | Descr | Value
-- | -- | --
Model Name | tiny-yolov2
Model Type | Convolutional Neural Network
Number of Layers | 25
Number of Input Layers | 1
Number of Output Layers | 0
Number of Convolutional Layers | 9
Number of Pooling Layers | 6
Number of Fully Connected Layers | 0
Number of Batch Normalization Layers | 8
Number of Weight Parameters | 11021232
Number of Bias Parameters | 5088
Total Number of Model Parameters | 11026320
Approximate Memory Cost for Training (MB) | 11533

**Comment by sasxianhu** [comment]
Can you paste your training log?

**Comment by mdziec** [comment]
| Descr | Value
-- | -- | --
Model Name | tiny-yolov2
Model Type | Convolutional Neural Network
Number of Layers | 25
Number of Input Layers | 1
Number of Output Layers | 0
Number of Convolutional Layers | 9
Number of Pooling Layers | 6
Number of Fully Connected Layers | 0
Number of Batch Normalization Layers | 8
Number of Weight Parameters | 11021232
Number of Bias Parameters | 5088
Total Number of Model Parameters | 11026320
Approximate Memory Cost for Training (MB) | 11533

**Comment by sasxianhu** [comment]
This is just the model info. What is your iteration history?

Best,
XQ

On Dec 8, 2018, at 7:47 PM, mdziec <notifications@github.com<mailto:notifications@github.com>> wrote:


EXTERNAL

        Descr   Value
Model Name      tiny-yolov2
Model Type      Convolutional Neural Network
Number of Layers        25
Number of Input Layers  1
Number of Output Layers 0
Number of Convolutional Layers  9
Number of Pooling Layers        6
Number of Fully Connected Layers        0
Number of Batch Normalization Layers    8
Number of Weight Parameters     11021232
Number of Bias Parameters       5088
Total Number of Model Parameters        11026320
Approximate Memory Cost for Training (MB)       11533

**Comment by bensloane** [comment]
When you run fit() you should have a log of loss and fit error per epoch.

**Comment by mdziec** [comment]
I have corrected the log above

**Comment by bensloane** [comment]
Your learning rate seems very high to begin with. I would lower the initial learning rate to start. It could be that you are updating with too large a step and never converging.

**Comment by sasxianhu** [comment]
The pasted log is misaligned. The learning rate is 0.001. But the training loss is still too big and the fit error is not improved at all. Something seems wrong. Is your input data correct?

**Comment by bensloane** [comment]
Oh the table is off by a column I'm seeing that now. Not a problem with the LR.

**Comment by mdziec** [comment]
Do you have access to RACE? If so, please send an email to splmdz@spl.sas.com

**Comment by bensloane** [comment]
I can follow up on this XQ

**Comment by sasxianhu** [comment]
Okay. Thanks.

Best,
XQ

On Dec 8, 2018, at 7:56 PM, Ben Sloane <notifications@github.com<mailto:notifications@github.com>> wrote:


EXTERNAL

I can follow up on this XQ

**Comment by bensloane** [comment]
Issue resolved can be closed. 

Over fit on training set produced valid predictions during inference on training data.

**Comment by sasxianhu** [comment]
Overfitting a small set of training data is always a good sign. :)
Thanks, Ben. Close it now.

**Comment by jackyyanggp** [comment]
Hi @bensloane @sasxianhu, could you please provide more details about how the issue was solved? did you change any parameters when "fitting"? I am running into the same issues. Thanks!
